client/src/components/cityview/realm/RealmNpcComponent.tsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useMemo, useState } from "react";
+import { Tabs } from "../../../elements/tab";
+import { NpcPanel } from "./villagers/NpcPanel";
+import useRealmStore from "../../../hooks/store/useRealmStore";
+import useUIStore from "../../../hooks/store/useUIStore";
+import { useRoute, useLocation } from "wouter";
+
+type RealmVillagersComponentProps = {};
+
+export const RealmNpcComponent = ({}: RealmVillagersComponentProps) => {
+  const [selectedTab, setSelectedTab] = useState(0);
+  const { realmEntityId } = useRealmStore();
+
+  const moveCameraToLaborView = useUIStore((state) => state.moveCameraToLaborView);
+  const moveCameraToFoodView = useUIStore((state) => state.moveCameraToFoodView);
+
+  // @ts-ignore
+  const [location, setLocation] = useLocation();
+  // @ts-ignore
+  const [match, params]: any = useRoute("/realm/:id/:tab");
+
+  useEffect(() => {
+    let _tab: string = "";
+    if (["farmers"].includes(params?.tab as string)) {
+      //   _tab = "farmers";
+      moveCameraToFoodView();
+    } else {
+      //   _tab = params?.tab as any;
+      moveCameraToLaborView();
+    }
+    const tabIndex = tabs.findIndex((tab) => tab.key === _tab);
+    if (tabIndex >= 0) {
+      setSelectedTab(tabIndex);
+    }
+  }, [params]);
+
+  const tabs = useMemo(
+    () => [
+      {
+        key: "villagers",
+        label: <div>All</div>,
+        component: <NpcPanel />,
+      },
+      //   {
+      //     key: "farmers",
```

Comment:
why is this commented out?

Reply:
Old relic of the Game Jam, I'll remove that